ckan/config/middleware/flask_app.py

```diff
@@ -340,6 +356,28 @@ def set_remote_user_as_current_user_for_tests():
             session["_user_id"] = userobj.id
 
 
+def add_csrf_token_to_session():
+    from itsdangerous import URLSafeTimedSerializer
+
+    login_blueprint = "user.login"
+    register_blueprint = "user.register"
+    is_login_form = request.endpoint == login_blueprint
+    is_register_form = request.endpoint == register_blueprint
+
+    # WTF_CSRF_FIELD_NAME//TIME_LIMIT are set by flask_wtf
+    field_name = config.get_value("WTF_CSRF_FIELD_NAME")
+    time_limit = config.get_value("WTF_CSRF_TIME_LIMIT")
+
+    if not session.get(field_name) and (is_login_form or is_register_form):
+        secret_key = config.get_value("SECRET_KEY")
```

Comment:
Shouldn't we try to use `WTF_CSRF_SECRET_KEY` first?

Reply:
I commented out this function, I think we might not need this anymore, I tested it for 2 hours and it works as it is supposed to.
Previously the csrf_token was not added in the session when login/register and it was failing with 400: Session token is missing. 
That's why at first I made this helper function
It works now without it, can you test it and confirm, please?

@pdelboca

Reply:
we can delete it later if everything's okay